src/item.cpp

```diff
@@ -5512,30 +5621,38 @@ bool item::is_power_armor() const
     return t->power_armor;
 }
 
-int item::get_encumber( const Character &p, encumber_flags flags ) const
+int item::get_encumber( const Character &p, const bodypart_id &bodypart,
+                        encumber_flags flags ) const
 {
     const islot_armor *t = find_armor_data();
     if( t == nullptr ) {
         // handle wearable guns (e.g. shoulder strap) as special case
         return is_gun() ? volume() / 750_ml : 0;
     }
-    int encumber = t->encumber;
+
+    int encumber = 0;
 
     // Additional encumbrance from non-rigid pockets
     float relative_encumbrance = 1;
     if( !( flags & encumber_flags::assume_full ) ) {
         relative_encumbrance = contents.relative_encumbrance();
     }
 
-    encumber += std::ceil( relative_encumbrance * ( t->max_encumber - t->encumber ) );
+    for( const random_armor_data &data : t->data ) {
+        if( bodypart == bodypart_id( bodypart_str_id( "num_bp" ) )
```

Comment:
What is this intended to do when passed `num_bp`?  It's unclear to me.

Reply:
Thanks for this spot, this check (and the following code that set the flag in item_factory) are now no longer needed. It was old legacy handling code, before I refactored the way legacy items are handled.